### ISSUE TITLE
iOS "Unable to load image file" on implicitly referenced project

### ISSUE THREAD

**Issue by Lonswaya** [dotnet/maui#21469]
### Description

We ran into a particular issue referencing MauiImage resource through multiple projects. 

We have a `Views` project referencing an `Images` project, and the `Views` project was referenced by the main project to display common components. 

When using views from the `Images` project or directly using images from the `Images` project, the main project failed to display the images in iOS only. Android displayed the images without issue. 

In addition, an exception is shown in the debug output: 

```
System.InvalidOperationException: Unable to load image file.
   at Microsoft.Maui.FileImageSourceService.GetImageAsync(IFileImageSource imageSource, Single scale, CancellationToken cancellationToken)
```

Our previous implementation used EmbeddedResources render images, converting SVGs into PNGs. As part of the Xamarin-to-Maui migration, we changed the image resources to `MauiImage` types and immediately observed this issue. 

I've attached a sample project with a simple implementation of the issue. 

![screenshot](https://github.com/dotnet/maui/assets/8558830/98c2316f-a7f2-4171-a05d-19cc08460092)

### Steps to Reproduce

1. Create a new MAUI solution
2. Create another MAUI project named "Views"
3. Create another MAUI project named "Images"
4. Create a SVG or a PNG image in "Images"
5. In the "Images" project, create a dependency to "Views" 
6. Create a view in "Views" using an image in "Images"
7. In the main project, create a dependency to "Views"
8. In the main project, reference the view in "Views" and display it on your main page. 
9. Run the application in iOS
10. EXPECTED: the image from "Images" is visible within the view. ACTUAL: The image from "Images" is not visible. 

### Link to public reproduction project repository

https://github.com/Lonswaya/MauiIOSImageIssue

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16.2 and up

### Did you find any workaround?

One alternative workaround is to turn an embedded resource into an image stream, and use `ImageSource.FromStream` to create a `StreamImageSource`. 

Like Maui, this has to be converted from an SVG into a PNG, so I have only implemented it for a PNG. You will need to use another library like SkiaSharp to turn a SVG into a PNG before using the resource. 

You can see a sample implementation in my reproduction project: https://github.com/Lonswaya/MauiIOSImageIssue/blob/227d0d10a85983dfce8795cbb98733a6c325c67c/TestMaui/MainPage.xaml.cs#L19

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio 17.10.0 preview2 (8.0.14). Can repro on iOS platform with sample project. 
https://github.com/Lonswaya/MauiIOSImageIssue

**Comment by PureWeen** [comment]
@Lonswaya for now if your app references all your projects does that fix?

**Comment by Lonswaya** [comment]
@PureWeen Yes, when explicitly referencing the library containing missing images, I am able to render images correctly. This is another potential workaround!

**Comment by riccardominato** [comment]
Same issue here. @PureWeen workarounds seems to work.
